dagql/backends/camflow/__init__.py

```diff
@@ -0,0 +1,120 @@
+import io
+import json
+import logging
+import subprocess
+from collections import defaultdict, deque
+
+from dagql.backend import Backend, Node, Edge, TraversalOrder
+
+
+NAMESPACE = b'dagql'
+START_DIRECTIVE = NAMESPACE + b': #start'
+NAMESPACE_PREFIX = NAMESPACE + b': '
+
+
+class CamFlowNode(Node):
+    """DAGQL node representation"""
+    def __init__(self, attrdict: dict):
+        self.attrs = attrdict
+
+    def __getitem__(self, attr: str):
+        return self.attrs[attr.lower()]
+
+
+class CamFlowEdge(Edge):
+    """DAGQL edge representation"""
+    def __init__(self, attrdict: dict):
+        self.attrs = attrdict
+
+    def __getitem__(self, attr: str):
+        return self.attrs[attr.lower()]
+
+
+class CamFlowBackend(Backend):
+    """
+    DAGQL backend for querying CamFlow provenance graphs.
+
+    Requires the provenance2json CamQuery hook to be loaded.
+    """
+    def __init__(self, msgbuf=None):
+        if msgbuf is None:
+            result = subprocess.run(['dmesg', '-t'], stderr=subprocess.DEVNULL,
+                                    stdout=subprocess.PIPE)
+            msgbuf = io.BytesIO(result.stdout)
+
+        self._reset_graph()
+
+        warned = False
+        for line in msgbuf:
+            if line.startswith(START_DIRECTIVE):
```

Comment:
Just a precaution: what happens if there is `irrelevant: #start` somewhere in between the lines of `dagql:`? will it reset the graph?

Maybe we should test for the full string `dagql: #start`, not just `#start`?

Reply:
And seems like for every msgbuf, we only have to reset the graph once (there is only one `#start` call in every input file). Why bother re-initializing again?

Reply:
`START_DIRECTIVE` _is_ defined as `"dagql: #start"`:

```python
NAMESPACE = b'dagql'
START_DIRECTIVE = NAMESPACE + b': #start'
```